limitador/src/storage/redis/redis_cached.rs

```diff
@@ -155,21 +168,23 @@ impl AsyncCounterStorage for CachedRedisStorage {
         }
 
         // Batch or update depending on configuration
-        if self.batching_is_enabled {
+        if self.is_partitioned() || self.batching_is_enabled {
             let mut batcher = self.batcher_counter_updates.lock().unwrap();
             for counter in counters.iter() {
-                match batcher.get_mut(counter) {
-                    Some(val) => {
-                        *val += delta;
-                    }
-                    None => {
-                        batcher.insert(counter.clone(), delta);
-                    }
-                }
+                Self::batch_counter(delta, &mut batcher, counter);
             }
         } else {
             for counter in counters.iter() {
-                self.update_counter(counter, delta).await?
+                self.update_counter(counter, delta).await.or_else(|err| {
+                    if err.is_transient() {
+                        self.partitioned(true);
+                        let mut batcher = self.batcher_counter_updates.lock().unwrap();
+                        Self::batch_counter(delta, &mut batcher, counter);
```

Comment:
We are just collecting and updating the counters in memory only to this point, right?

Reply:
Not sure what you mean... that's when they are flushed to redis... yes. Always was the case btw, this isn't much different from before.

Reply:
then that batch should be sent to redis once we recover the connection, right? I reckon we don't consider the case the previous redis instance died for good

Reply:
So @didierofrivia did spot an issue here, we might be partitioned and _not having batching enabled_ which would result in the batch size growing on every partition without ever flushing to Redis... 

I'm thinking we should _always_ enable batching as: 
 - It optimizes most for the happy path (i.e. limits not reached)
 - We are planning on using this implementation for Global RL, which adds such a big latency on going to redis... 

Thoughts?

Reply:
I agree with you that batching should be always enabled 👍🏼